### PR TITLE
Allow player data to be read from file system

### PR DESCRIPTION
**Pull request by Mongey**
This keeps the current way of storing in typescript files, but, this also allows you to have a json file in `public/players/my-mates.json` and then visiting http://localhost:4200/#/my-mates  ... so I don't need to commit them

### CODE REVIEW

File: app/routes/worldcup.ts [hunk 1]
```diff
@@ -2,31 +2,34 @@ import Route from '@ember/routing/route';
...
 import { data as ashData } from 'worldcup/data/ashbourne';
 
-type SweepstakesId = 'college' | 'pintmen' | 'babb' | 'ashbourne';
+var fixedData = {
+  college: collegeData,
+  pintmen: pintmenData,
+  babb: babb,
+  ashbourne: ashData,
+};
+
```
Comment (nolaneo) [inline app/routes/worldcup.ts:17]:
```suggestion
const fixedData: Record<SweepstakesId, Array<Player>> = {
  college: collegeData,
  pintmen: pintmenData,
  babb: babb,
  ashbourne: ashData,
};

```

File: app/routes/worldcup.ts [hunk 2]
```diff
@@ -2,31 +2,34 @@ import Route from '@ember/routing/route';
...
-      this.sweepstakes.setPlayers(pintmenData as Array<Player>);
-    }
-    if (params.id === 'babb') {
-      this.sweepstakes.setPlayers(babb as Array<Player>);
-    }
-    if (params.id === 'ashbourne') {
-      this.sweepstakes.setPlayers(ashData as Array<Player>);
+  async model(params: { id: string }) {
+    //@ts-ignore
+    var players = fixedData[params.id] || null;
```
Comment (nolaneo) [inline app/routes/worldcup.ts:24]:
```suggestion
    let players = fixedData[params.id] || null;
```

File: app/routes/worldcup.ts [hunk 3]
```diff
@@ -2,31 +2,34 @@ import Route from '@ember/routing/route';
...
+    if (players === null) {
+      // attempt to load from the file system
+      let response = await fetch(`/players/${params.id}.json`);
+      let loadedData = await response.json();
+      players = loadedData.data;
```
Comment (nolaneo) [inline app/routes/worldcup.ts:29]:
```suggestion
      players = loadedData.data as Array<Player>;
```

**Comment by Mongey** [comment]
@nolaneo I just migrated the current files over to the json way data files

**Comment by Mongey** [comment]
Can now generate a set of players with `ruby generators/world_cup_generator.rb foo,bar,moo,cow blah`
and then visit http://localhost:4200/#/blah

**Pull request merged**